### PR TITLE
Refactor History Panel Code a bit

### LINKED ISSUE #13127: History page throws error when selecting one of my areas
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

After selecting one of areas in history the page is just showing a spinner and in console I get error:
ha-panel-history.ts:209 Uncaught (in promise) TypeError: i is not iterable
    at r.value (ha-panel-history.ts:209:1)
    at r.value (ha-panel-history.ts:209:1)
    at r.value (ha-panel-history.ts:209:1)
    at r._$AE (styles.ts:369:1)
    at r.performUpdate (styles.ts:369:1)
    at r.scheduleUpdate (styles.ts:369:1)
    at r._$EC (styles.ts:369:1)
    at async r._$EC (styles.ts:369:1)

### Describe the behavior you expected

Should load history

### Steps to reproduce the issue

1. go to history
2. select specific area
3. page is loading all the time
4. error in console
...

### What version of Home Assistant Core has the issue?

2022.7.0

### What was the last working version of Home Assistant Core?

2022.6.x

### In which browser are you experiencing the issue with?

Chrome 103

### Which operating system are you using to run this browser?

macOS Monterey

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

```txt
ha-panel-history.ts:209 Uncaught (in promise) TypeError: i is not iterable
    at r.value (ha-panel-history.ts:209:1)
    at r.value (ha-panel-history.ts:209:1)
    at r.value (ha-panel-history.ts:209:1)
    at r._$AE (styles.ts:369:1)
    at r.performUpdate (styles.ts:369:1)
    at r.scheduleUpdate (styles.ts:369:1)
    at r._$EC (styles.ts:369:1)
    at async r._$EC (styles.ts:369:1)
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Cleaning the code up a bit
Fixes: #13127 
Removes some redundant code

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/history/ha-panel-history.ts [hunk 1]
```diff
@@ -413,18 +339,20 @@ class HaPanelHistory extends SubscribeMixin(LitElement) {
...
-              entityIds.add(foundEntity.entity_id);
-            }
+        const foundEntities = this._deviceEntityLookup[singleSearchingDeviceId];
+        if (!foundEntities?.length) {
+          continue;
+        }
+
+        for (const foundEntity of foundEntities) {
+          if (foundEntity.entity_category === null) {
+            entityIds.add(foundEntity.entity_id);
```
Comment (D3v01dZA) [inline src/panels/history/ha-panel-history.ts:349]: This line can cause a separate issue I saw, if you hit return after typing an invalid entity_id the target picker will accept that value. Ideally we should check that this entity_id actually exists, and unfortunately that means you may want to reintroduce the _stateEntities state value to be sure that you're adding a valid entity_id.

Comment (zsarnett) [reply src/panels/history/ha-panel-history.ts:349]: We should fix this in the picker

AllowCustomValue should be used to allow or not allow

File: src/panels/history/ha-panel-history.ts [hunk 2]
```diff
@@ -391,19 +312,24 @@ class HaPanelHistory extends SubscribeMixin(LitElement) {
...
-            for (const foundDeviceEntity of foundDeviceEntities) {
-              if (
-                (!foundDeviceEntity.area_id ||
-                  foundDeviceEntity.area_id === singleSearchingAreaId) &&
-                foundDeviceEntity.entity_category === null
-              ) {
-                entityIds.add(foundDeviceEntity.entity_id);
-              }
+        const foundDevices = this._areaDeviceLookup[singleSearchingAreaId];
+        if (!foundDevices) {
```
Comment (D3v01dZA) [inline src/panels/history/ha-panel-history.ts:316]: Most of the time you're checking for !x?.length, any reason to do this instead? Technically the length is not necessary anywhere, it should just remove the need to create an empty iterator at each for loop, but my OCD made me type this comment :)

File: src/panels/history/ha-panel-history.ts (hunk 1 above)
Comment (D3v01dZA) [reply src/panels/history/ha-panel-history.ts:349]: Sorry, actually that's line #430 that'g not correct.

File: src/panels/history/ha-panel-history.ts (hunk 2 above)
Comment (zsarnett) [reply src/panels/history/ha-panel-history.ts:316]:
```suggestion
        if (!foundDevices.length) {
```

Comment (D3v01dZA) [reply src/panels/history/ha-panel-history.ts]: I rarely use Github so I'm not familiar with suggesting changes, but this should be !foundDevices?.length

File: src/panels/history/ha-panel-history.ts [hunk 3]
```diff
@@ -391,19 +312,24 @@ class HaPanelHistory extends SubscribeMixin(LitElement) {
...
-            for (const foundDeviceEntity of foundDeviceEntities) {
-              if (
-                (!foundDeviceEntity.area_id ||
-                  foundDeviceEntity.area_id === singleSearchingAreaId) &&
-                foundDeviceEntity.entity_category === null
-              ) {
-                entityIds.add(foundDeviceEntity.entity_id);
-              }
+        const foundDevices = this._areaDeviceLookup[singleSearchingAreaId];
+        if (!foundDevices.length) {
```
Comment (D3v01dZA) [inline src/panels/history/ha-panel-history.ts:316]:
```suggestion
        if (!foundDevices?.length) {
```

**Comment by krzysztofsk** [comment]
I'm not sure if it was covered here but it would be good to hide already added items from the lookup or prevent from adding it multiple times as it only adds another pill but doesn't show multiple graphs

**Review by bramkragten — APPROVED** [review]

**Pull request merged**